src/history.h

```diff
@@ -40,9 +40,13 @@ struct io_streams_t;
 typedef std::vector<wcstring> path_list_t;
 
 enum history_search_type_t {
-    // The history searches for strings containing the given string.
+    // We don't know what type of search to perform.
+    HISTORY_SEARCH_TYPE_UNKNOWN,
```

Comment:
Done. Although it does make the option parsing slightly less clean IMHO. What I shouldn't have done was explicitly test for that enum in *history.cpp* and instead just relied on the default clause to catch that it hadn't been set to a known value. Which is also why I really dislike using the default value of zero for the first enum symbol. It makes it too easy for an uninitialized enum to be treated as if it was set to the first symbol.